### PR TITLE
Show notification on kill

### LINKED ISSUE #318: When app get's killed, like dismissed it should throw a notification asking to reopen like tesla app does ($100)
**Is your feature request related to a problem? Please describe.**
The app has to be opened all the time in order for the device to work, could be in the background but not killed.

**Describe the solution you'd like**
When this happens there should be an event that detects this and throws a notification, asking to reopen.
There should be an explanation about it during the onboarding.

### PR DESCRIPTION
**Pull request by Becca-Saka**
Triggers a notification when the app is terminated 

iOS

https://github.com/BasedHardware/Friend/assets/47469638/fe80b453-c74c-47c5-8ad2-4953308807cf

Android 

https://github.com/BasedHardware/Friend/assets/47469638/9fff51c1-5237-4a67-9d81-78e45de402b2

Fixes #318

### CODE REVIEW

**Comment by josancamon19** [comment]
Awesome, as per the approach I see, you built the code for android and iOS right, and are calling it through flutter.
Weren't there any libraries available?

Also, could you please use linting rule of line character length of 120 + default `dart format` / android studio default rules.

**Comment by Becca-Saka** [comment]
There are libraries but they haven't been updated in a while. 

I will fix this lint issue and tag you again.

> Awesome, as per the approach I see, you built the code for android and iOS right, and are calling it through flutter. Weren't there any libraries available?
> 
> Also, could you please use linting rule of line character length of 120 + default `dart format` / android studio default rules.

**Comment by Becca-Saka** [comment]
@josancamon19 fixed the lint issue.

**Review by josancamon19 — APPROVED** [review]

File: apps/AppWithWearable/ios/Runner/AppDelegate.swift [hunk 1]
```diff
@@ -3,13 +3,38 @@ import UIKit
...
+
+    //Creates a method channel to handle notifications on kill
+    let controller = window?.rootViewController as? FlutterViewController
+    methodChannel = FlutterMethodChannel(name: "com.friend.ios/notifyOnKill", binaryMessenger: controller!.binaryMessenger)
+    methodChannel?.setMethodCallHandler { [weak self] (call, result) in
+      self?.handleMethodCall(call, result: result)
+    }
+
+    // Request notification permission
+    UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound, .badge]) { granted, error in
```
Comment (josancamon19) [inline apps/AppWithWearable/ios/Runner/AppDelegate.swift:28]: Same here, let's not request notifications right when it opens.

File: apps/AppWithWearable/android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -55,6 +55,7 @@
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
```
Comment (josancamon19) [inline apps/AppWithWearable/android/app/src/main/AndroidManifest.xml:55]: Doesn't seem to be reliable on android.
And let's not ask for permissions right when the app opens.

https://github.com/BasedHardware/Friend/assets/18078879/c1edc2cc-3257-4250-9897-5187959df398

**Review by josancamon19 — CHANGES_REQUESTED** [review]

**Comment by josancamon19** [comment]
Also let's set the notification content to:
title: Friend Device Disconnected
body: Please keep your app opened to continue using your Friend.

**Comment by Becca-Saka** [comment]
> Also let's set the notification content to: title: Friend Device Disconnected body: Please keep your app opened to continue using your Friend.

Got it, I will work on the requested changes tomorrow.

**Comment by josancamon19** [comment]
Hi @Becca-Saka are there any updates to this?

**Comment by josancamon19** [comment]
I will go ahead and merge this as it is, because it is useful. But looking forward for the remaining changes, thanks!

**Comment by Becca-Saka** [comment]
Hi @josancamon19  Sorry for the delayed response. I removed the notification request on the start-up

I created a new PR here https://github.com/BasedHardware/Friend/pull/355

I couldn't replicate the Android issues, you raised though.



https://github.com/BasedHardware/Friend/assets/47469638/03820210-f9ed-4f2a-89a8-152e59ab39de

**Comment by Becca-Saka** [comment]
We should probably move our conversation to the new PR?

**Pull request merged**